From 78ae6314245b0e53322a64abdd8ef9f2ad00e4b4 Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Mon, 18 Nov 2024 21:24:57 +0100
Subject: [PATCH 05/10] Make default permissions updates tied to version
 incremental

Otherwise adding new rules won't have any effect for us.

Change-Id: Ib81082a184956ea550bdccdbdf53b2b4cc48f487
---
 .../com/android/server/pm/PackageManagerService.java   | 10 ++++++----
 .../pm/permission/PermissionManagerServiceImpl.java    |  3 ++-
 2 files changed, 8 insertions(+), 5 deletions(-)

diff --git a/services/core/java/com/android/server/pm/PackageManagerService.java b/services/core/java/com/android/server/pm/PackageManagerService.java
index 0e343e18ab34..3b48266af8a8 100644
--- a/services/core/java/com/android/server/pm/PackageManagerService.java
+++ b/services/core/java/com/android/server/pm/PackageManagerService.java
@@ -4275,7 +4275,7 @@ public class PackageManagerService implements PackageSender, TestUtilityService
             final int userId = livingUsers.get(i).id;
             final boolean isPermissionUpgradeNeeded = !Objects.equals(
                     mPermissionManager.getDefaultPermissionGrantFingerprint(userId),
-                    Build.FINGERPRINT);
+                    Build.VERSION.INCREMENTAL);
             if (isPermissionUpgradeNeeded) {
                 grantPermissionsUserIds = ArrayUtils.appendInt(
                         grantPermissionsUserIds, userId);
@@ -4284,7 +4284,8 @@ public class PackageManagerService implements PackageSender, TestUtilityService
         // If we upgraded grant all default permissions before kicking off.
         for (int userId : grantPermissionsUserIds) {
             mLegacyPermissionManager.grantDefaultPermissions(userId);
-            mPermissionManager.setDefaultPermissionGrantFingerprint(Build.FINGERPRINT, userId);
+            mPermissionManager.setDefaultPermissionGrantFingerprint(
+                    Build.VERSION.INCREMENTAL, userId);
         }
         if (grantPermissionsUserIds == EMPTY_INT_ARRAY) {
             // If we did not grant default permissions, we preload from this the
@@ -4466,7 +4467,8 @@ public class PackageManagerService implements PackageSender, TestUtilityService
         if (!convertedFromPreCreated || !readPermissionStateForUser(userId)) {
             mPermissionManager.onUserCreated(userId);
             mLegacyPermissionManager.grantDefaultPermissions(userId);
-            mPermissionManager.setDefaultPermissionGrantFingerprint(Build.FINGERPRINT, userId);
+            mPermissionManager.setDefaultPermissionGrantFingerprint(
+                    Build.VERSION.INCREMENTAL, userId);
             mDomainVerificationManager.clearUser(userId);
         }
     }
@@ -4478,7 +4480,7 @@ public class PackageManagerService implements PackageSender, TestUtilityService
             mPermissionManager.readLegacyPermissionStateTEMP();
             final boolean isPermissionUpgradeNeeded = !Objects.equals(
                     mPermissionManager.getDefaultPermissionGrantFingerprint(userId),
-                    Build.FINGERPRINT);
+                    Build.VERSION.INCREMENTAL);
             return isPermissionUpgradeNeeded;
         }
     }
diff --git a/services/core/java/com/android/server/pm/permission/PermissionManagerServiceImpl.java b/services/core/java/com/android/server/pm/permission/PermissionManagerServiceImpl.java
index ea71953d7cb3..b420e29d86ab 100644
--- a/services/core/java/com/android/server/pm/permission/PermissionManagerServiceImpl.java
+++ b/services/core/java/com/android/server/pm/permission/PermissionManagerServiceImpl.java
@@ -4644,7 +4644,8 @@ public class PermissionManagerServiceImpl implements PermissionManagerServiceInt
     @Nullable
     @Override
     public String getDefaultPermissionGrantFingerprint(@UserIdInt int userId) {
-        return mPackageManagerInt.isPermissionUpgradeNeeded(userId) ? null : Build.FINGERPRINT;
+        return mPackageManagerInt.isPermissionUpgradeNeeded(userId)
+                ? null : Build.VERSION.INCREMENTAL;
     }
 
     @Override
-- 
2.34.1

